python_scripts/parameter_tuning_randomized_search.py

```diff
@@ -238,66 +255,17 @@ def shorten_param(param_name):
 cv_results = pd.read_csv("../figures/randomized_search_results.csv",
                          index_col=0)
 
-# %% [markdown]
-# As we have more than 2 parameters in our grid-search, we cannot visualize the
-# results using a heatmap. However, we can us a parallel coordinates plot.
-
-# %%
 (cv_results[column_results].rename(
-    shorten_param, axis=1).sort_values("mean_test_score"))
-
-# %%
-import numpy as np
-import plotly.express as px
-
-fig = px.parallel_coordinates(
-    cv_results.rename(shorten_param, axis=1).apply({
-        "learning_rate": np.log10,
-        "max_leaf_nodes": np.log2,
-        "max_bins": np.log2,
-        "min_samples_leaf": np.log10,
-        "l2_regularization": np.log10,
-        "mean_test_score": lambda x: x}),
-    color="mean_test_score",
-    color_continuous_scale=px.colors.sequential.Viridis,
-)
-fig.show()
+    shorten_param, axis=1).sort_values("mean_test_score", ascending=False))
 
 # %% [markdown]
-#
-# The parallel coordinates plot will display the values of the hyperparameters
-# on different columns while the performance metric is color coded. Thus, we
-# are able to quickly inspect if there is a range of hyperparameters which is
-# working or not.
-#
-# ```{note}
-# We **transformed most axis values by taking a log10 or log2** to
-# spread the active ranges and improve the readability of the plot.
-# ```
-#
-# In particular for this hyper-parameter search, it is interesting to see that
-# the yellow lines (top performing models) all reach intermediate values for
-# the learning rate, that is, tick values between -2 and 0 which correspond to
-# learning rate values of 0.01 to 1.0 once we invert the log10 transform for
-# that axis.
-#
-# It is possible to **select a range of results by clicking and holding on any
-# axis** of the parallel coordinate plot. You can then slide (move) the range
-# selection and cross two selections to see the intersections. You can undo a
-# selection by clicking once again on the same axis.
-#
-# We also observe that it is not possible to select the highest performing
-# models by selecting lines of on the `max_bins` axis with tick values between
-# 1 and 3.
-#
-# The other hyper-parameters are not very sensitive. We can check that if we
-# select the `learning_rate` axis tick values between -1.5 and -0.5 and
-# `max_bins` tick values between 5 and 8, we always select top performing
-# models, whatever the values of the other hyper-parameters.
```

Comment:
Are you sure you want to remove this from `python_scripts/parameter_tuning_randomized_search.py `?

If you add it only to `python_scripts/video_parallel_plot.py `, this notebook will no longer be reachable from the navigation menus.

I think it's import ant to keep it as part of the interactive notebooks from the module. I would insert the video in the sequence right after `python_scripts/parameter_tuning_randomized_search.py` so that readers:

- first see the interactive plots and play with it intuitively but probably miss how to use it correctly,
- then watch the video;
- the go back to the interactive plot to check they can select a range by clicking on the axis.

Reply:
Or alternatively we can keep the split in two notebooks and insert the page with the new video between them but please adjust the `_toc.yml` file as part of this PR so that we have the big picture before merging the PR.

Reply:
Thanks, done!